### ISSUE TITLE
Kiosk restarted several times

### ISSUE THREAD

**Issue by john-deng** [loft-sh/kiosk#129]
Kiosk restarted several times with below error

Version: kiosksh/kiosk:0.2.4

```bash
➜  ~ k get po -l app=kiosk
NAME                     READY   STATUS    RESTARTS   AGE
kiosk-6bbff86fd4-gts8n   1/1     Running   9          40d

➜  ~ k logs -p kiosk-6bbff86fd4-gts8n
E0513 03:51:13.342092       1 deleg.go:144] setup: unable to create client Get "https://10.80.0.1:443/api?timeout=32s": dial tcp 10.80.0.1:443: connect: connection refused
```

**Comment by FabianKramm** [comment]
@john-deng thanks for creating this issue! Seems like your cluster was not available during that time or kiosk couldn't connect to it, what Kubernetes cluster are you using?

**Comment by john-deng** [comment]
@FabianKramm I am using kind cluster which is running Kubernetes v1.20.2, kiosk works fine, just saw the pod restarted so often, it may not work during the pod restarting I guess. so there is a risk that the kiosk will not available from time to time. Is the logging that I pasted above helps? It has only one line of logging presence when I type kubectl logs -p kiosk-pod-name.

**Comment by FabianKramm** [comment]
@john-deng yes the log message helps, however I'm afraid there is not much we can do on our side to avoid this, as kiosk requires interaction with the Kubernetes API server during startup.
